hnn_core/network_models.py

```diff
@@ -0,0 +1,237 @@
+    loc = 'soma'
+    receptor = 'ampa'
+    net._all_to_all_connect(
+        src_cell, target_cell, loc, receptor, nc_dict)
+
+    src_cell = 'L2Pyr'
+    key = f'gbar_L2Pyr_{target_cell}'
+    nc_dict['A_weight'] = net._params[key]
+    loc = 'soma'
+    receptor = 'ampa'
+    net._all_to_all_connect(
+        src_cell, target_cell, loc, receptor, nc_dict)
+
+    return net
+
+
+def beta_erp_network(params=None, add_drives_from_params=False):
+    """Instantiate the beta modulated ERP network model.
+
+    Parameters
+    ----------
+    params : dict | None
+        The parameters to use for constructing the network.
+        If None, parameters loaded from beta_erp.json
+        Default: None
+    add_drives_from_params : bool
+        If True, add drives as defined in the params-dict. NB this is mainly
+        for backward-compatibility with HNN GUI, and will be deprecated in a
+        future release. Default: False
+
+    Returns
+    -------
+    net : Instance of Network object
+        Network object used to store the default network.
+        All connections defining the default network will be
+        appeneded to net.connectivity.
+    Notes
+    -----
+    Model reproduces results from Law et al. 2021
+    This model differs from the default network model in several
+    parameters including: (fill in later)
+    """
+
+    hnn_core_root = op.dirname(hnn_core.__file__)
+    params_fname = op.join(hnn_core_root, 'param', 'default.json')
+    if params is None:
+        params = read_params(params_fname)
+
+    net = default_network(params)
+
+    # Update biophysics (increase gabab duration of inhibition)
+    net.cell_types['L2_pyramidal'].p_syn['gabab']['tau1'] = 45.0
+    net.cell_types['L2_pyramidal'].p_syn['gabab']['tau2'] = 200.0
+    net.cell_types['L5_pyramidal'].p_syn['gabab']['tau1'] = 45.0
+    net.cell_types['L5_pyramidal'].p_syn['gabab']['tau2'] = 200.0
+
+    # Decrease L5_pyramidal -> L5_pyramidal nmda weight
+    net.connectivity[2]['nc_dict']['A_weight'] = 0.0004
+
+    # Modify L5_basket -> L5_pyramidal inhibition
+    net.connectivity[6]['nc_dict']['A_weight'] = 0.02  # gabaa
+    net.connectivity[7]['nc_dict']['A_weight'] = 0.005  # gabab
+
+    # Remove L5 pyramidal somatic and basal dendrite calcium channels
+    for sec in ['soma', 'basal_1', 'basal_2', 'basal_3']:
+        net.cell_types['L5_pyramidal'].p_secs[
+            sec]['mechs']['ca']['gbar_ca'] = 0.0
+
+    # Remove L2_basket -> L5_pyramidal gabaa connection
+    del net.connectivity[10]  # Original paper simply sets gbar to 0.0
+
+    # Add L2_basket -> L5_pyramidal gabab connection
+    nc_dict = {
+        'A_delay': net.delay,
+        'threshold': net.threshold,
+    }
+    src_cell = 'L2Basket'
+    target_cell = 'L5Pyr'
+    nc_dict['lamtha'] = 50.
+    nc_dict['A_weight'] = 0.0002
+    loc = 'distal'
+    receptor = 'gabab'
+    net._all_to_all_connect(
+        src_cell, target_cell, loc, receptor, nc_dict)
```

Comment:
it's a bit odd that we're using this private method when the public method is `add_connection`. Could we modify the API of `add_connection` so it can be used in this situation? Maybe an argument called `drop_autapses` which defaults to `False` ?

Reply:
Just realized that this isn't even necessary since `net.add_connection` defaults to all-to-all with `allow_autapses=True`.

Honestly if I can find a clean way to allow `unique=True` connections for the drives, then we can get rid of `_all_to_all_connect`

Reply:
Great! I think we should move towards using `net.add_connection` in our internal API as much as possible -- e.g., in `default_network` by using `drop_autapses` argument. Then we can get rid of the method once we figure out what to do about `unique=True`

Reply:
I think I've figured it out! Now to rid the world of `all_to_all_connect` once and for all....